proposals/NNNN-debug-printf.md

```diff
@@ -0,0 +1,56 @@
+# Dxil Debug Printf
+
+* Proposal: [NNNN](NNNN-debug-printf.md)
+* Author(s): [Jiao Lu](https://github.com/jiaolu)
+* Sponsor: TBD
+* Status: **Under Consideration**
+
+## Introduction
+
+Add new dxil op to allow c/c++ like *printf* intrinsic instructions can be generated in the dxil sourced from hlsl.
+
+## Motivation
+
+As the new shader models, i.e. RayTracing, RayQuery, Workgraph, and more complex algorithms and structure emerges in recent time, we meet many more requirements of debugging capabilities of hlsl coding in game, application and driver development.
+
+The dxil counterpart spirv has a similar feature, [NonSemantic.DebugPrintf extention](https://github.com/KhronosGroup/SPIRV-Registry/blob/main/nonsemantic/NonSemantic.DebugPrintf.asciidoc) to generate DebugPrintf spirvOp souced from hlsl/glsl. Based on the spirvOp extention, some vendor drivers and Valve lunarG has [debug printf layer](https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/main/docs/debug_printf.md) to dump printf expression, "hlsl/glsl variables" into stdio or file.
+
+## Proposed solution
+
+The printf expression in hlsl mostly like this example.
+```c++ hlsl:
+
+const string str0= "str0";
+string str1 = "str1";
+
+void main() {
+    printf(str0);
+    printf(str1);
+    printf("Variables are: %d %d %.2f", 1u, 2u, 1.5f);
```

Comment:
If we decide to go further with this, we'll need to specify exactly how these format strings are interpreted.

Reply:
dxc does not valiate format specifier to the c/c++ format speicifer standard, or the matching relation between
format specifier and argument. If the number and type don't match, they will produce undefined result from 
client api, e.g. driver. It's client api responsibility to output printf string.

Reply:
I don't think we want to build a feature that DXC doesn't validate and where all behavior is undefined.

Reply:
this is part of design has been changed.